### DISCUSSION TITLE
Get xls datetime as number

### DISCUSSION THREAD

**Discussion by topmet** [PHPOffice/PhpSpreadsheet#3525] [Q&A] [answered]
get all data from worksheet as  ..... ->toArray(null, true, true, true);

Cell in xls file "01.03.2023 10:57:06" get in  ->toArray(null, true, true, true);   "44986.456319444"

How to fix it ?

O

**Comment by MarkBaker** [comment]
This should work: what version of PhpSpreadsheet are you using?

Or are you loading a file and telling the Reader to read Data Only?

**Reply by topmet** [reply]
1.28.0 
php 7.4

`$inputFileType = 'Xls';
$inputFileName = $A['file']['xls'];


$reader = \PhpOffice\PhpSpreadsheet\IOFactory::createReader($inputFileType);

$reader->setReadDataOnly(true);

$spreadsheet = $reader->load($inputFileName);
$sheetData = $spreadsheet->getActiveSheet()->toArray(null, true, true, true);`

can't load file (

**Comment by MarkBaker** [comment, marked as answer]
You do realise that `$reader->setReadDataOnly(true);` is telling the Reader to load only the raw data from the worksheet, and not to load the formatting information for cells?

The value that looks like `01.03.2023 10:57:06` in the Excel cell is an Excel serialized timestamp, a numeric value (`44986.456319444`) with a number format mask that tells Excel to display it like a date.
When you do `$reader->setReadDataOnly(true);`, you're telling PhpSpreadsheet to load the value (`44986.456319444`), but **_not_** to load the number format mask.

See [the documentation here](https://phpspreadsheet.readthedocs.io/en/latest/topics/reading-files/#reading-only-data-from-a-spreadsheet-file)

**Reply by topmet** [reply]
omg, thanks, work fine
